src/funnels/automation/StableSwapper.sol

```diff
@@ -16,15 +16,22 @@
 
 pragma solidity ^0.8.16;
 
+import "./utils/EnumerableSet.sol";
+
 interface SwapperLike {
     function swap(address, address, uint256, uint256, address, bytes calldata) external returns (uint256);
 }
 
 contract StableSwapper {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
     mapping (address => uint256) public wards;                           // Admins
     mapping (address => uint256) public buds;                            // Whitelisted keepers
     mapping (address => mapping (address => PairConfig)) public configs; // Configuration for keepers
+    mapping (bytes32 => Pair) public pairs;
```

Comment:
Some early comments:

1. If we go to this direction of using a bytes32 as the identifier in the enumerable set I would be in favor of using it also as the key for `configs` (and thus removing the nested mappings).
From a check I did the hash calculation would add 80 gas units to the swap, which I think is ok (Uniswap does the same when storing the position - https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/Position.sol#L36).
If we do that then it might be nice to also make the hash logic public so it's easy to access `configs` of a pair.

2. I'd move `pairs` to be adjacent to `pairHashes`, so it's clear these are just used as utilities for iterating configs.

3. I think it's more standard to use encodePacked for the hashing (especially if is saves memory slots in some of the contracts).

Reply:
All done

Reply:
Nice, seems like this even improves gas, at least in some cases.

Checked the first deposit in `testDepositWithdrawByKeeper` on block 17868830:
Before both commits                - gasUsed 269251
After `Use hash as config key` - gasUsed 268187
After `Use abi.encodePacked` - gasUsed 268108